libs/core/tests/unit_tests/prompts/test_chat.py

```diff
@@ -831,6 +831,41 @@ def test_chat_prompt_message_placeholder_tuple() -> None:
         assert optional_prompt.format_messages() == []
 
 
+def test_chat_prompt_message_placeholder_dict() -> None:
+    prompt = ChatPromptTemplate([{"role": "placeholder", "content": "{convo}"}])
+    assert prompt.format_messages(convo=[("user", "foo")]) == [
+        HumanMessage(content="foo")
+    ]
+
+    assert prompt.format_messages() == []
+
+    # Is optional = True
+    optional_prompt = ChatPromptTemplate(
+        [{"role": "placeholder", "content": ["{convo}", False]}]
+    )
+    assert optional_prompt.format_messages(convo=[("user", "foo")]) == [
+        HumanMessage(content="foo")
```

Comment:
for content as a list this might want to be `content=["foo"]` because list templates i believe are supported - might want a case with 2 elements in the list

Reply:
ignore me this is a placeholder